dataprofiler/profilers/graph_profiler.py

```diff
@@ -390,3 +392,74 @@ def _attribute_data_as_list(self, graph, attribute):
             value = graph[u][v][attribute]
             data_as_list.append(value)
         return data_as_list
+
+    def _save_helper(self, filepath, data_dict):
+        """
+        Save profiler to disk.
+
+        :param filepath: Path of file to save to
+        :type filepath: String
+        :param data_dict: profile data to be saved
+        :type data_dict: dict
+        :return: None
+        """
+        # Set Default filepath
+        if filepath is None:
+            filepath = "profile-{}.pkl".format(
+                datetime.now().strftime("%d-%b-%Y-%H:%M:%S.%f")
+            )
+
+        # add profiler class to data_dict
+        data_dict["profiler_class"] = self.__class__.__name__
+
+        # Pickle and save profile to disk
+        with open(filepath, "wb") as outfile:
+            pickle.dump(data_dict, outfile)
+
+    def save(self, filepath=None):
+        """
+        Save profiler to disk.
+
+        :param filepath: Path of file to save to
+        :type filepath: String
+        :return: None
+        """
+        # Create dictionary for all metadata, options, and profile
+        data_dict = {
+            "sample_size": self.sample_size,
+            "times": self.times,
+            "_attributes": self._attributes,
+            "_num_nodes": self._num_nodes,
+            "_num_edges": self._num_edges,
+            "_categorical_attributes": self._categorical_attributes,
+            "_continuous_attributes": self._continuous_attributes,
+            "_avg_node_degree": self._avg_node_degree,
+            "_global_max_component_size": self._global_max_component_size,
+            "_continuous_distribution": self._continuous_distribution,
+            "_categorical_distribution": self._categorical_distribution,
+            "metadata": self.metadata,
+            "__calculations": self.__calculations,
+        }
+
+        self._save_helper(filepath, data_dict)
+
+    @classmethod
+    def load(cls, filepath):
+        """
+        Load profiler from disk.
+
+        :param filepath: Path of file to load from
+        :type filepath: String
+        :return: GraphProfiler being loaded
+        :rtype: GraphProfiler
+        """
+        # Load profile from disk
+        with open(filepath, "rb") as infile:
+            data = pickle.load(infile)
+
+        profiler = cls(None, options=None)
```

Comment:
just any class? should there not be some logic around loading the correct class for the profiler?

Reply:
cls is the class it is loading within, this is same logic in BaseProfiler which will allow us to merge all 3 profilers in the long term.